FrontBundle/Routing/Database/OpenOrchestraDatabaseUrlGenerator.php

```diff
@@ -0,0 +1,109 @@
+    protected $routeDocumentToValueObjectTransformer;
+    const REDIRECT_TO_LANGUAGE = 'redirect_to_language';
+
+    /**
+     * Constructor.
+     *
+     * @param RouteDocumentRepositoryInterface      $routeDocumentRepository
+     * @param RouteDocumentToValueObjectTransformer $routeDocumentToValueObjectTransformer
+     * @param RequestStack                          $requestStack
+     * @param NodeManager                           $nodeManager
+     * @param RequestContext                        $context The context
+     * @param LoggerInterface|null                  $logger  A logger instance
+     */
+    public function __construct(
+        RouteDocumentRepositoryInterface $routeDocumentRepository,
+        RouteDocumentToValueObjectTransformer $routeDocumentToValueObjectTransformer,
+        RequestStack $requestStack,
+        NodeManager $nodeManager,
+        RequestContext $context,
+        LoggerInterface $logger = null)
+    {
+        $this->routeDocumentToValueObjectTransformer = $routeDocumentToValueObjectTransformer;
+        $this->routeDocumentRepository = $routeDocumentRepository;
+        $this->request = $requestStack->getMasterRequest();
+        $this->nodeManager = $nodeManager;
+        $this->context = $context;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Generates a URL or path for a specific route based on the given parameters.
+     *
+     * @param string $name The name of the route
+     * @param mixed $parameters An array of parameters
+     * @param bool|string $referenceType The type of reference to be generated (one of the constants)
+     *
+     * @return string The generated URL
+     *
+     * @throws RouteNotFoundException              If the named route doesn't exist
+     * @throws MissingMandatoryParametersException When some parameters are missing that are mandatory for the route
+     * @throws InvalidParameterException           When a parameter value for a placeholder is not correct because
+     *                                             it does not match the requirement
+     */
+    public function  generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
+    {
+        if (isset($parameters[self::REDIRECT_TO_LANGUAGE])) {
+            try {
+                $fullName = $this->nodeManager->getNodeRouteName($name, $parameters[self::REDIRECT_TO_LANGUAGE]);
+            } catch (NodeNotFoundException $e) {
+                throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', $name));
+            }
+            unset($parameters[self::REDIRECT_TO_LANGUAGE]);
+        } else {
+            $aliasId = 0;
+            if ($this->request) {
+                $aliasId = $this->request->get('aliasId', $aliasId);
+            }
+
+            $fullName = $aliasId . '_' . $name;
+        }
+
+        $routeDocument = $this->routeDocumentRepository->findOneByName($fullName);
```

Comment:
The site id is not required because the route name (the only information needed to generate a route) is composed by the aliasId and the mongo node Id.

The alias Id is held by the incoming request (through the matching)